apps/builder/app/builder/features/settings-panel/resource-panel.tsx

```diff
@@ -51,7 +54,104 @@ import {
   composeFields,
 } from "~/shared/form-utils";
 import { ExpressionEditor } from "~/builder/shared/expression-editor";
-import { parseCurl } from "./curl";
+import { parseCurl, type CurlRequest } from "./curl";
+
+const UrlField = ({
+  scope,
+  aliases,
+  value,
+  onChange,
+  onCurlPaste,
+}: {
+  aliases: Map<string, string>;
+  scope: Record<string, unknown>;
+  value: string;
+  onChange: (value: string) => void;
+  onCurlPaste: (curl: CurlRequest) => void;
+}) => {
+  const urlId = useId();
+  const ref = useRef<HTMLTextAreaElement>(null);
+  const [error, setError] = useState("");
```

Comment:
not sure but I think you could use html as well as a way to set error message

Reply:
of course I could, but I need to render it in tooltip, not only style when it exists